drivers/macvlan/macvlan_joinleave.go

```diff
@@ -39,38 +39,49 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		return fmt.Errorf("could not find endpoint with id %s", eid)
 	}
 	// parse and match the endpoint address with the available v4 subnets
-	if len(n.config.Ipv4Subnets) > 0 {
-		s := n.getSubnetforIPv4(ep.addr)
-		if s == nil {
-			return fmt.Errorf("could not find a valid ipv4 subnet for endpoint %s", eid)
+	if !n.config.Internal {
+		if len(n.config.Ipv4Subnets) > 0 {
```

Comment:
the log messages below can by de-duplicated by writing
if len(n.config.Ipv4Subnets) > 0 && !n.config.Internal with a comment saying why we are skipping this
same goes for Ipv6

Reply:
Those log messages are not duplicates. The later one is missing _Gateway: %s_ part.